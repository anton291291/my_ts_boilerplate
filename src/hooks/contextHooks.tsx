import React, { useState, createContext } from 'react';
export const GridContext = createContext({} as GridContextType);

export declare namespace GlobalState {
    type CellItemType = { index: number; isClicked: boolean };

    type AxisType = {
        axis: {
            x: number;
            y: number;
        };
    };

    type CellType = {
        cells: Array<CellItemType>;
    };

    type IntervalIDType = {
        intervalID: number;
    };

    type isPlayType = {
        isPlay: boolean;
        speed: number;
    };

    type FormType = {
        name: string;
    };

    type GenType = {
        gen: number;
    };
}

export type StateType = GlobalState.AxisType &
    GlobalState.CellType &
    GlobalState.IntervalIDType &
    GlobalState.GenType &
    GlobalState.isPlayType &
    GlobalState.FormType;

type GridContextType = {
    state: StateType;
    setState: React.Dispatch<React.SetStateAction<StateType>>;
};

export const GridProvider = ({ children }) => {
    const y = 15;
    const x = 30;

    const [state, setState] = useState({
        axis: { x: x, y: y },
        cells: Array.from({ length: y * x }, (item, index) => ({
            index: index,
            isClicked: false
        })),
        gen: 1,
        name: '',
        intervalID: null,
        isPlay: false,
        speed: 1
    });

    return (
        <GridContext.Provider value={{ state, setState }}>
            {children}
        </GridContext.Provider>
    );
};
