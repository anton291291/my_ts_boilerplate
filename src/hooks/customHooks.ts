const useLogin = (params) => {
const [state, setstate] = useState(initialState)
}