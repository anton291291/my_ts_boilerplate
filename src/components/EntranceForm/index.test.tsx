import React from 'react';
import { mount, shallow } from 'enzyme';
import { mountToJson, shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { EntranceForm } from '.';
import { GridProvider } from '../../hooks/contextHooks';

describe('EntranceForm', () => {
    const output = mount(
        <EntranceForm />,

        {
            wrappingComponent: GridProvider
        }
    );

    test('should render correctly', () => {
        expect(mountToJson(output)).toMatchSnapshot();
    });

    test('EntranceBtn onClick func  should open Modal', () => {
        output.find('EntranceBtn').simulate('click');
        expect(output.find('ForwardRef(Modal)').prop('open')).toBeTruthy();
    });

    test('Modal onClose prop should close it', () => {
        output.find('ForwardRef(Modal)').invoke('onClose')();
        expect(output.find('ForwardRef(Modal)').prop('open')).toBeFalsy();
    });

});
