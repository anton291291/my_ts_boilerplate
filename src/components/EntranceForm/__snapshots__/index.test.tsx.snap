// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntranceForm should render correctly 1`] = `
<EntranceForm>
  <EntranceBtn
    onClick={[Function]}
  >
    <ForwardRef
      className="EntranceBtn-qvz05a-0 YjlRU"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(SvgIcon))
        className="EntranceBtn-qvz05a-0 YjlRU"
        onClick={[Function]}
      >
        <ForwardRef(SvgIcon)
          className="EntranceBtn-qvz05a-0 YjlRU"
          classes={
            Object {
              "colorAction": "MuiSvgIcon-colorAction",
              "colorDisabled": "MuiSvgIcon-colorDisabled",
              "colorError": "MuiSvgIcon-colorError",
              "colorPrimary": "MuiSvgIcon-colorPrimary",
              "colorSecondary": "MuiSvgIcon-colorSecondary",
              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
              "root": "MuiSvgIcon-root",
            }
          }
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root EntranceBtn-qvz05a-0 YjlRU"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </ForwardRef(SvgIcon)>
      </WithStyles(ForwardRef(SvgIcon))>
    </ForwardRef>
  </EntranceBtn>
  <ForwardRef(Modal)
    aria-describedby="simple-modal-description"
    aria-labelledby="simple-modal-title"
    onClose={[Function]}
    open={false}
  />
</EntranceForm>
`;
